src/inspect_ai/model/_call_tools.py

```diff
@@ -165,7 +165,7 @@ async def call_tool_task(
             # create event
             event = ToolEvent(
                 id=call.id,
-                function=call.function,
+                function=call.native_name or call.function,
```

Comment:
This is the backward transformation pattern. It works, but it feels a little wonky. It seems error prone since consumers will have to be mindful of the transformation concept and understand if they want the name of the inspect function actually called, or the name the model thought they wanted called.

Happy to discuss, or have you take a pass at it.